### PR TITLE
Upgrade telemetry - add new events

### PR DESCRIPTION
**Pull request by kiiadi**
Add new events for credential type and region id.

### CODE REVIEW

File: src/credentials/providers/sharedCredentialsProvider.ts [hunk 1]
```diff
@@ -73,17 +79,24 @@ export class SharedCredentialsProvider implements CredentialsProvider {
     public validate(): string | undefined {
         const expectedProperties: string[] = []
 
-        if (this.hasProfileProperty(SHARED_CREDENTIAL_PROPERTIES.ROLE_ARN)) {
+        if (
+            this.hasProfileProperty(SHARED_CREDENTIAL_PROPERTIES.ROLE_ARN) &&
+            this.hasProfileProperty(SHARED_CREDENTIAL_PROPERTIES.SOURCE_PROFILE)
```
Comment (awschristou) [inline src/credentials/providers/sharedCredentialsProvider.ts:84]: This change isn't tied to the telemetry changes. Please add it to the commit comments and PR description in case we need to look back at the commit history.

File: src/test/credentials/provider/sharedCredentialsProvider.test.ts [hunk 2]
```diff
@@ -108,6 +108,16 @@ describe('SharedCredentialsProvider', async () => {
         assertSubstringsInText(sut.validate(), MISSING_PROPERTIES_FRAGMENT, 'aws_secret_access_key')
     })
 
+    it('validation identifies assume role must have source_profile', async () => {
```
Comment (justinmk3) [inline src/test/credentials/provider/sharedCredentialsProvider.test.ts:111]: the wording is hard to parse. suggestion:

```suggestion
    it('"assume role" must have source_profile', async () => {
```

Comment (kiiadi) [reply src/test/credentials/provider/sharedCredentialsProvider.test.ts:111]: I was kinda following the existing pattern - but happy to change ...
